source/application/AMPIC887DriverServer/AMPIC887AxisMap.h

```diff
@@ -0,0 +1,297 @@
+#ifndef AMPIC887AXISMAP_H
+#define AMPIC887AXISMAP_H
+
+#include <QMap>
+#include <QList>
+#include "AMGCS2.h"
+#include "AMPIC887AxisCollection.h"
+#include "AMGCS2Support.h"
+#include "util/AMCArrayHandler.h"
+/*!
+  * A class which maps Axes of a PI C887.11 controller to a type T.
+  */
+template <typename T>
+class AMPIC887AxisMap : public QMap<AMGCS2::Axis, T>
+{
+public:
+	/*!
+	  * Creates an instance of an axis map, which is empty.
+	  */
+	AMPIC887AxisMap() {}
+
+	/*!
+	  * Returns a collection of the axes which this map contains.
+	  */
+	AMPIC887AxisCollection axes() const
+	{
+		AMPIC887AxisCollection returnCollection(AMPIC887AxisCollection::EmptyCollection);
+		QList<AMGCS2::Axis> axisEntries = this->keys();
+
+		foreach(AMGCS2::Axis currentAxisEntry, axisEntries) {
```

Comment:
This one confused me a bit. 
You're right, the code you posted works, but what's confusing me is that there is no constructor in AMPIC887AxisColleciton which takes a type QList<AMGCS::Axis>.

Does this mean that you can always construct a derived class by passing a base instance to its constructor?